### PR TITLE
Fix SDL_AddTimer not honoring timeout argument

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #17941

### CODE REVIEW

File: test/browser/test_sdl2_timer.c [hunk 1]
```diff
@@ -34,10 +39,10 @@ int main(int argc, char** argv) {
   int badret = 4;
   int goodret = 5;
 
-  SDL_TimerID badtimer = SDL_AddTimer(500, report_result, &badret);
-  SDL_TimerID goodtimer = SDL_AddTimer(1000, report_result, &goodret);
+  Uint32 start_time = SDL_GetTicks();
```
Comment (kripken) [inline test/browser/test_sdl2_timer.c:42]:
```suggestion
  start_time = SDL_GetTicks();
```

File: test/browser/test_sdl2_timer.c [hunk 2]
```diff
@@ -10,13 +10,18 @@
...
+Uint32 SDLCALL timer_callback(Uint32 interval, void *param) {
   static int reported = 0;
-  if (reported) return 0;
+  assert(!reported);
   reported = 1;
-  SDL_Quit();
   int result = *(int *)param;
-  printf("report_result: %p %d\n", param, result);
+  printf("timer_callback: %p %d\n", param, result);
+  assert(result == 5);
```
Comment (kripken) [inline test/browser/test_sdl2_timer.c:21]: The bad timer with 4 has a timeout of 500 though, shouldn't it happen first?

File: test/browser/test_sdl2_timer.c [hunk 3]
```diff
@@ -10,13 +10,18 @@
...
+  Uint32 now = SDL_GetTicks();
+  assert(now > start_time + 500);
```
Comment (kripken) [inline test/browser/test_sdl2_timer.c:23]:
```suggestion
  assert(now >= start_time + 500);
```
Might avoid some type of rare flake.

File: test/browser/test_sdl2_timer.c (hunk 2 above)
Comment (sbc100) [reply test/browser/test_sdl2_timer.c:21]: The point of the bad timer is that it should never fire (since it gets cancelled before we return from main).

**Review by kripken — APPROVED** [review]

**Pull request merged**